### ISSUE TITLE
Default cache store not correctly documented

### ISSUE THREAD

**Issue by brchristian** [rails/rails#28535]
In a brand new Rails 5.0.2 application, the default cache store is `memory_store` (as you can see [here](https://github.com/rails/rails/blob/5-0-stable/railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt#L19)).

However, there are two problems with this. The first is that (unless there is something wrong with my setup) because the cache memory is not shared across processes, you cannot for instance clear the cache by typing `Rails.cache.clear` from the Rails console. This is unexpected behavior and hard to diagnose.

The second problem is that the [documentation guide for caching](https://github.com/rails/rails/blob/master/guides/source/caching_with_rails.md) says that `:file_store` is the default implementation. This is incorrect, as the default implementation is actually `:memory_store`.

In my opinion, `:file_store` makes more sense as the default implementation because it allows a server process and a console process to interact in what (to me) seems like the expected manner. But there may be other considerations.

So in conclusion, it seems to me that we would need a PR that _either_:

(a) Reverts the default to `:file_store`, _OR_

(b) Updates the documentation to correctly indicate that `:memory_store` is the default implementation.

I would be happy to submit either PR based on the preferred direction of those who might know more about this than I.

**Comment by rafaelfranca** [comment]
[The default is the :file_store](https://github.com/rails/rails/blob/49c92cc661f60afafd44db8cc8b6d61f647ff174/railties/lib/rails/application/configuration.rb#L43), unless you explicitly set that, like it is being done in `development.rb` of the application. The default in that guide talk about the default when no configuration is set.

About the default in the generated app being `memory_store`, it is a good question. When the feature was suggested in https://github.com/rails/rails/issues/18875 it was already the default. @dhh do you know why the memory store is the default in development.rb?

**Comment by matthewd** [comment]
We default other things (AJ, ACa) to in-process backends in development too. `memory_store` seems consistent in that regard.

I could actually see myself making an argument that it should be `memory_store` *and* it should be automatically cleared every time we do a class reload...

**Comment by dhh** [comment]
Agree. Memory store all the way for dev.

**Comment by brchristian** [comment]
Sounds good. I’m personally not clear on the motivations behind `:memory_store` in dev, but I defer to everyone’s much greater expertise here. 👍

So it sounds like what might be handy then is to add some language to the guides detailing a couple points:

1. `:file_store` is "default" in the sense that it will be selected if there is no explicit configuration saying otherwise; however `:memory_store` is explicitly made the "default" config (in dev environment) for new Rails projects.

2. When using `:memory_store`, the lack of a shared resource across processes means that you can’t read/write/expire the cache through the Rails console. Therefore new users in particular who spin up fresh Rails projects and attempt to follow along with a tutorial (e.g., http://railscasts.com/episodes/90-fragment-caching-revised) on caching won’t be confused (as I was at first) when the console commands don’t work.

I’ve prepared a PR (https://github.com/rails/rails/issues/28547) to this effect. Let me know what you think.

Cheers!

**Comment by maclover7** [comment]
Closing since #28547, which resolves this issue, has been merged.
